plugins/modules/synchronize.py

```diff
@@ -619,15 +633,20 @@ def _write_password_to_pipe(proc):
     out_lines = out_clean.split('\n')
     while '' in out_lines:
         out_lines.remove('')
-    if module._diff:
-        diff = {'prepared': out_clean}
-        return module.exit_json(changed=changed, msg=out_clean,
-                                rc=rc, cmd=cmdstr, stdout_lines=out_lines,
-                                diff=diff)
 
-    return module.exit_json(changed=changed, msg=out_clean,
-                            rc=rc, cmd=cmdstr, stdout_lines=out_lines)
+    result = dict(changed=changed, rc=rc, cmd=cmdstr)
+
+    if quiet:
+        result['msg'] = "OUTPUT IS HIDDEN DUE TO 'quiet=true'"
+        result['stdout_lines'] = []
+    else:
+        result['msg'] = out_clean
+        result['std_out'] = out_lines
```

Comment:
based on @phemmer feedback, this could rather be:
```suggestion
    result = dict(changed=changed, rc=rc, cmd=cmdstr, stdout_lines=out_lines, msg=out_clean)
    if quiet:
        result['msg'] = "%s files/directories have been synchronized" % len(out_lines)
```

Reply:
This seems better, though I was wondering if len(out_lines) would give the actual number of files/directories syncd?